pkg/client/build.go

```diff
@@ -1063,7 +1077,51 @@ func prependBuildpackToOrder(order dist.Order, bpInfo dist.ModuleInfo) (newOrder
 	return newOrder
 }
 
-func (c *Client) createEphemeralBuilder(rawBuilderImage imgutil.Image, env map[string]string, order dist.Order, buildpacks []buildpack.BuildModule) (*builder.Builder, error) {
+func (c *Client) processExtensions(ctx context.Context, builderImage imgutil.Image, builderExs []dist.ModuleInfo, builderOrder dist.Order, stackID string, opts BuildOptions) (fetchedExs []buildpack.BuildModule, orderExtensions dist.Order, err error) {
+	relativeBaseDir := opts.RelativeBaseDir
+	declaredExs := opts.Extensions
+
+	orderExtensions = dist.Order{{Group: []dist.ModuleRef{}}}
+	for _, ex := range declaredExs {
+		locatorType, err := buildpack.GetLocatorType(ex, relativeBaseDir, builderExs)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		switch locatorType {
+		case buildpack.RegistryLocator:
+			return nil, nil, errors.New("RegistryLocator type is not valid for extensions")
+		case buildpack.FromBuilderLocator:
+			switch {
+			case len(orderExtensions) == 0 || len(orderExtensions[0].Group) == 0:
+				orderExtensions = builderOrder
+			case len(orderExtensions) > 1:
+				// This should only ever be possible if they are using from=builder twice which we don't allow
+				return nil, nil, errors.New("extensions from builder can only be defined once")
+			default:
+				newOrderExtensions := dist.Order{}
+				groupToAdd := orderExtensions[0].Group
+				for _, bOrderEntry := range builderOrder {
+					newEntry := dist.OrderEntry{Group: append(groupToAdd, bOrderEntry.Group...)}
+					newOrderExtensions = append(newOrderExtensions, newEntry)
+				}
```

Comment:
```suggestion
			return nil, nil, errors.New("from builder is not supported for extensions")
```
Apologies, my comment above was a bit unclear. I think we just want to error in this case. 

(We'd also need to delete the three lines below - 1108-1110 currently - but I can't include them in my suggestion because of Codecov's comment)